src/lib/viewers/error/PreviewErrorViewer.js

```diff
@@ -91,8 +94,11 @@ class PreviewErrorViewer extends BaseViewer {
         this.iconEl.innerHTML = this.icon;
         this.messageEl.textContent = displayMessage;
 
-        // Add optional download button
-        if (checkPermission(file, PERMISSION_DOWNLOAD) && showDownload && Browser.canDownload()) {
+        // Add optional link or download button
+        const { linkMsg, linkUrl } = err;
```

Comment:
Where are we adding this linkUrl to the error object?

Reply:
In a separate patch when the field is ready.

Reply:
Ignore what I said, I've added the check for `is_download_available` into this patch now.